### PR TITLE
#HT-166: Create models and tests for `Conversation` and `Comment`

### PR DESCRIPTION
**Pull request by pranjal5399**

### CODE REVIEW

File: packages/common/source/types/index.ts [hunk 1]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    updatedAt: Date;
+}
+
+export interface Coordinates {
+    x: Number;
+    y: Number;
+}
+export interface Conversation {
+    /* An identifier uniquely identifies the conversation across Hypertool. */
+    id: string;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:432]: ObjectId. The attribute is supposed to be `_id`, not `id`.

File: packages/common/source/types/index.ts [hunk 2]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
     createdAt: Date;
     updatedAt: Date;
 }
+
+export interface Comment {
+    /* An identifier that uniquely identifies the comment across Hypertool. */
+    id: string;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:398]: The attribute name is supposed to be `_id`. The type is `ObjectId`.

File: packages/common/source/types/index.ts [hunk 3]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+export interface Coordinates {
+    x: Number;
+    y: Number;
+}
+export interface Conversation {
+    /* An identifier uniquely identifies the conversation across Hypertool. */
+    id: string;
+
+    /* An identifier that points to the App where the comment was created. */
+    app: App;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:435]: `ObjectId | App`

File: packages/common/source/types/index.ts [hunk 4]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+
+    /* The name of the Page where the comment was created. */
+    page: string | Page;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:438]: `ObjectId | Page`

File: packages/common/source/types/index.ts [hunk 5]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+
+    /* An object that describes the x and y coordinates of the conversation in
+     * the canvas.
+     */
+    coordinates: Coordinates;
+
+    /* A list of users who have participated in the conversation. */
+    taggedUsers: string[] | [User];
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:446]: `ObjectId[] | User[]`

File: packages/common/source/types/index.ts [hunk 6]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+
+    /* A list of comments in the conversation. The first member is the
+     * initiator’s comment.
+     */
+    comments: string[] | [Comment];
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:451]: `ObjectId[] | Comment[]`

File: packages/common/source/types/index.ts [hunk 7]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+
+    /* A enumeration of string values that describes the status of the
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:453]: An*

File: packages/common/source/types/index.ts [hunk 8]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    createdAt: Date;
+
+    /* Specifies the timestamp that indicates when the conversation was last
+     * modified
+     */
+    updatedAt: Date;
+}
+
+export interface Page {
+    id: string;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:468]: The attribute name should be `_id`. Type should be `ObjectId`.

File: packages/common/source/types/index.ts [hunk 9]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+
+    /* An identifier that points to the App where the comment was created. */
+    app: string | App;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:471]: `ObjectId | App`

File: packages/common/source/types/index.ts [hunk 10]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    /* The slug of the page. */
+    slug: string;
+
+    /* Specifies the timestamp that indicates when the page was created */
+    createdAt: Date;
+
+    /* Specifies the timestamp that indicates when the page was last modified */
+    updatedAt: Date;
+}
+export interface ExternalConversation {
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:488]: Line break.

File: packages/common/source/types/index.ts [hunk 11]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    id: string;
+    app: App;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:490]: `string | App`

File: packages/common/source/types/index.ts [hunk 12]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    page: string | Page;
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:491]: `string`

File: packages/common/source/types/index.ts [hunk 13]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    coordinates: Coordinates;
+    taggedUsers: string[] | [User];
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:493]: `string[] | User[]`

File: packages/common/source/types/index.ts [hunk 14]
```diff
@@ -389,3 +392,107 @@ export interface ExternalMembership {
...
+    comments: string[] | [Comment];
```
Comment (itssamuelrowe) [inline packages/common/source/types/index.ts:494]: `string[] | Comment[]`

File: packages/common/test/helper/constants.ts [hunk 15]
```diff
@@ -116,6 +116,35 @@ const memberships = [
     },
 ];
 
+const pages = [
+    {
+        app: "61c93a931da4a79d3a109947",
+        title: "Page-1",
+        description: "This is the first page",
+        slug: "slug content",
```
Comment (itssamuelrowe) [inline packages/common/test/helper/constants.ts:124]: `/apps/123/builder`

File: packages/common/test/helper/constants.ts [hunk 16]
```diff
@@ -116,6 +116,35 @@ const memberships = [
     },
 ];
 
+const pages = [
+    {
+        app: "61c93a931da4a79d3a109947",
+        title: "Page-1",
```
Comment (itssamuelrowe) [inline packages/common/test/helper/constants.ts:122]: App Builder

**Review by itssamuelrowe — CHANGES_REQUESTED** [review]

File: packages/common/source/models/PageModel.ts [hunk 17]
```diff
@@ -0,0 +1,43 @@
+import { Schema, model } from "mongoose";
+
+import type { Page } from "../types";
+
+const pageSchema = new Schema(
+    {
+        /* An identifier that points to the App where the comment was created. */
```
[the remark this answers is not in the document]
Comment (pranjal5399) [reply packages/common/source/models/PageModel.ts:7]: copied from docs

**Review by itssamuelrowe — APPROVED** [review]

**Review by joelerego — APPROVED** [review]

**Pull request merged**